cumulusci/tasks/tests/test_datadictionary.py

```diff
@@ -0,0 +1,559 @@
+                        os.path.sep, task.zip_prefix, "src/objects/Parent__c.object"
+                    ).strip(os.path.sep)
+                ),
+            ]
+        )
+
+        task._process_object_element.assert_has_calls(
+            [
+                call("Child__c", ET.fromstring("<test></test>"), LooseVersion("1.1")),
+                call("Parent__c", ET.fromstring("<test></test>"), LooseVersion("1.1")),
+            ]
+        )
+
+    @patch("zipfile.ZipFile")
+    def test_walk_releases__mdapi(self, zip_file):
+        task = create_task(
+            GenerateDataDictionary,
+            {"object_path": "object.csv", "field_path": "fields.csv"},
+        )
+        task.project_config.project__git__prefix_release = "rel/"
+
+        task.get_repo = Mock()
+        release = Mock()
+        release.draft = False
+        release.prerelease = False
+        release.tag_name = "rel/1.1"
+        task.get_repo.return_value.releases.return_value = [release]
+        task._process_mdapi_release = Mock()
+        zip_file.return_value.namelist.return_value = ["PREFIX/src/objects"]
+
+        task._walk_releases()
+
+        task._process_mdapi_release.assert_called_once_with(
+            zip_file.return_value, "1.1"
+        )
+
+    @patch("zipfile.ZipFile")
+    def test_walk_releases__sfdx(self, zip_file):
+        task = create_task(
+            GenerateDataDictionary,
+            {"object_path": "object.csv", "field_path": "fields.csv"},
+        )
+        task.project_config.project__git__prefix_release = "rel/"
+
+        task.get_repo = Mock()
+        release = Mock()
+        release.draft = False
+        release.prerelease = False
+        release.tag_name = "rel/1.1"
+        task.get_repo.return_value.releases.return_value = [release]
+        task._process_sfdx_release = Mock()
+        zip_file.return_value.namelist.return_value = [
+            "PREFIX/force-app/main/default/objects"
+        ]
+
+        task._walk_releases()
+
+        task._process_sfdx_release.assert_called_once_with(zip_file.return_value, "1.1")
+
+    def test_init_schema(self):
+        task = create_task(GenerateDataDictionary, {})
+        task._init_schema()
+
+        assert task.schema is not None
+
+    @patch("zipfile.ZipFile")
+    def test_run_task(self, zip_file):
+        # This is an integration test. We mock out `get_repo()` and the filesystem.
+        xml_source = """<?xml version="1.0" encoding="UTF-8"?>
+<CustomObject xmlns="http://soap.sforce.com/2006/04/metadata">
+    <description>Description</description>
+    <label>Test</label>
+    <fields>
+        <fullName>Type__c</fullName>
+        <inlineHelpText>Type of field.</inlineHelpText>
+        <label>Type</label>
+        <type>Text</type>
+    </fields>
+</CustomObject>"""
+        task = create_task(GenerateDataDictionary, {})
+        task.project_config.keychain.get_service = Mock()
+        task.project_config.init_sentry = Mock()
```

Comment:
Is there a less-monkeypatchy, more-recommended way to mock this out?

Reply:
hmm, init_sentry should exit right away anyway as long as the keychain doesn't have a sentry service configured. So maybe just don't mock keychain.get_service?

I need to redo all the sentry stuff at some point, since we're using the raven lib instead of sentry-sdk which is now preferred.